Server/controllers/monitorController.js

```diff
@@ -262,6 +263,43 @@ const createMonitor = async (req, res, next) => {
   }
 };
 
+/**
+ * Checks if the endpoint can be resolved and adds it to the job queue.
+ * @async
+ * @param {Object} req - The Express request object.
+ * @property {Object} req.query - The query parameters of the request.
+ * @param {Object} res - The Express response object.
+ * @param {function} next - The next middleware function.
+ * @returns {Object} The response object with a success status, a message, and the resolution result.
+ * @throws {Error} If there is an error during the process, especially if there is a validation error (422).
+ */
+
+const checkEndpointResolution = async (req, res, next) => {
+  try {
+    let { monitorURL } = req.query;
+
+    // Remove the protocol (http:// or https://) if present
+    monitorURL = monitorURL.replace(/^https?:\/\//, '');
+    // Remove the trailing slash if it exists
+    monitorURL = monitorURL.endsWith('/') ? monitorURL.slice(0, -1) : monitorURL;
+
+    dns.setServers(['8.8.8.8', '1.1.1.1']);// Google, Cloudfare    
+    dns.resolve(monitorURL, (error) => {
+      console.log("Inside .resolve");
+      if (error) {
+        next(handleError(error, SERVICE_NAME, "checkEndpointResolution"));
+        return ;
+      }
```

Comment:
By throwing the error, the server is getting crashed because 'The error thrown inside the callback function is not caught by the surrounding try-catch block because it happens asynchronously'.

Instead can we return the response directly?

if (error) {
        return res.status(400).json({
          success: false,
          msg: \`DNS resolution failed: ${error.message}\`,
        });
      }

Reply:
All error handling is done centrally via middleware, we do not want to have error handling done in multiple places.

You can promisify `dns.resolve` to throw the error:

```
await new Promise((resolve, reject) => {
	dns.resolve(monitorURL, (error) => {
		if (error) {
			reject(error);
		}
		resolve();
	});
});
return res.status(200).json({
	success: true,
	msg: `URL resolved successfully`,
});
```